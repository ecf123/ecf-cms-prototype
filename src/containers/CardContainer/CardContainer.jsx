import { FALSE } from 'sass'
import Card from '../../components/Card/Card'
import './CardContainer.scss'

const CardContainer = ({cards}) => {

    const renderCards = cards?.map((card, index) => {
        return (
            <Card 
                key = {index}
                img = {card.img}
                dateOrTime = {card.dateOrTime}
                title = {card.title} 
                cardInfo = {card.cardInfo}
                links = {card.links}
                condition={false}
            />
        )
    })
    
  return (
    <div className="card-container">{renderCards}</div>
  )
}

export default CardContainer