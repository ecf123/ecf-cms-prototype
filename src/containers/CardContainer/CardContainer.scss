@use '../../styles/variables.scss' as *;

.card-container {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(26.7rem, 1fr));
    border: 1px black solid;
    gap: 25px;
    justify-content: end;

    &__box {
        display: flex;
        width: 100%;
        height: 100%;
        justify-items: center;
        align-items: center;
    }
}