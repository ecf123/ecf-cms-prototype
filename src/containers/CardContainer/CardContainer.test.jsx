import {render, screen} from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import CardContainer from "./CardContainer";
import image from "../../assets/for-hire-image.svg"

const cards = [
  {img: image, dateOrTime: '12:01', title: 'Test title one', cardInfo: 'Test text one', links: ["one", "two", "three"]}, 
  {img: image, dateOrTime: '12:00', title: 'Test title two', cardInfo: 'Test text two', links: ["one", "two", "three"]}, 
];

it("it should match the snapshot with props", () => {
    const { container } = customRender(
      <CardContainer cards={cards} />
    );
    expect(container).toMatchSnapshot();
});

it("Should render the card with no props", () => {
    const { container } = customRender(<CardContainer />);
    expect(container).toMatchSnapshot();
});

it("should render the first card component on the screen", () => {
    render(<CardContainer cards = {cards}/>)
    const dateOrTime = screen.getByText("12:01");
    const title = screen.getByText("Test title one");
    const cardInfo = screen.getByText("Test text one");

    expect(dateOrTime).toBeInTheDocument();
    expect(title).toBeInTheDocument();
    expect(cardInfo).toBeInTheDocument();
});

it("should render the second card component on the screen", () => {
  render(<CardContainer cards = {cards}/>)
  const dateOrTime = screen.getByText("12:00");
  const title = screen.getByText("Test title two");
  const cardInfo = screen.getByText("Test text two");

  expect(dateOrTime).toBeInTheDocument();
  expect(title).toBeInTheDocument();
  expect(cardInfo).toBeInTheDocument();
});