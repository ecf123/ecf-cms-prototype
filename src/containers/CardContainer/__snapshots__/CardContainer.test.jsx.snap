// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the card with no props 1`] = `
<div>
  <div
    class="card-container"
  />
</div>
`;

exports[`it should match the snapshot with props 1`] = `
<div>
  <div
    class="card-container"
  >
    <div
      class="card"
    >
      <a
        href="/Testtitleone"
      >
        <img
          class="card__image"
          src="for-hire-image.svg"
        />
      </a>
      <p
        class="card__date"
      >
        12:01
      </p>
      <a
        class="card__header"
        href="/Testtitleone"
      >
        <h2
          class="card__title"
        >
          Test title one
        </h2>
        <img
          alt="Arrow Icon"
          class="card__icon"
          src="arrow-upper-right-black.svg"
        />
      </a>
      <p
        class="card__info"
      >
        Test text one
      </p>
      <ul
        class="categories-menu"
      >
        <li
          class="categories-menu__item categories-menu__item--purple"
        >
          one
        </li>
        <li
          class="categories-menu__item categories-menu__item--darkpurple"
        >
          two
        </li>
        <li
          class="categories-menu__item categories-menu__item--purple"
        >
          three
        </li>
      </ul>
    </div>
    <div
      class="card"
    >
      <a
        href="/Testtitletwo"
      >
        <img
          class="card__image"
          src="for-hire-image.svg"
        />
      </a>
      <p
        class="card__date"
      >
        12:00
      </p>
      <a
        class="card__header"
        href="/Testtitletwo"
      >
        <h2
          class="card__title"
        >
          Test title two
        </h2>
        <img
          alt="Arrow Icon"
          class="card__icon"
          src="arrow-upper-right-black.svg"
        />
      </a>
      <p
        class="card__info"
      >
        Test text two
      </p>
      <ul
        class="categories-menu"
      >
        <li
          class="categories-menu__item categories-menu__item--purple"
        >
          one
        </li>
        <li
          class="categories-menu__item categories-menu__item--darkpurple"
        >
          two
        </li>
        <li
          class="categories-menu__item categories-menu__item--purple"
        >
          three
        </li>
      </ul>
    </div>
  </div>
</div>
`;
